diff -rupN ROX-Filer-orig/src/menu.c ROX-Filer/src/menu.c
--- ROX-Filer-orig/src/menu.c	2011-03-03 17:55:26.000000000 +0200
+++ ROX-Filer/src/menu.c	2011-03-03 17:55:56.000000000 +0200
@@ -216,7 +216,7 @@ static GtkItemFactoryEntry filer_menu_de
 {">" N_("Delete"),	    	"<Ctrl>X", file_op, FILE_DELETE, "<StockItem>", GTK_STOCK_DELETE},
 {">",				NULL, NULL, 0, "<Separator>"},
 {">" N_("Shift Open"),   	NULL, file_op, FILE_OPEN_FILE},
-{">" N_("Send To..."),		NULL, file_op, FILE_SEND_TO, NULL},
+{">" N_("Open With..."),		NULL, file_op, FILE_SEND_TO, NULL},
 {">",				NULL, NULL, 0, "<Separator>"},
 {">" N_("Set Run Action..."),	"asterisk", file_op, FILE_RUN_ACTION, "<StockItem>", GTK_STOCK_EXECUTE},
 {">" N_("Set Icon..."),		NULL, file_op, FILE_SET_ICON, NULL},
