--- ROX-Filer/src/view_details.cORIG	2011-10-23 01:48:50.000000000 +0800
+++ ROX-Filer/src/view_details.c	2011-10-20 18:17:34.000000000 +0800
@@ -1059,26 +1059,39 @@
 	ADD_TEXT_COLUMN(_("_Name"), COL_LEAF);
 	gtk_tree_view_column_set_sort_column_id(column, COL_LEAF);
 	gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
+	
 	ADD_TEXT_COLUMN(_("_Type"), COL_TYPE);
 	gtk_tree_view_column_set_sort_column_id(column, COL_TYPE);
 	gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
+	
+	ADD_TEXT_COLUMN(_("_Size"), COL_SIZE);
+	g_object_set(G_OBJECT(cell), "xalign", 1.0, "font", "monospace", NULL);
+	g_signal_connect_after(object, "realize",
+			       G_CALLBACK(set_column_mono_font),
+			       G_OBJECT(cell));
+	gtk_tree_view_column_set_sort_column_id(column, COL_SIZE);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
+	
+	ADD_TEXT_COLUMN(_("Last _Modified"), COL_MTIME);
+	gtk_tree_view_column_set_sort_column_id(column, COL_MTIME);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
+	
 	ADD_TEXT_COLUMN(_("_Permissions"), COL_PERM);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
 	g_object_set(G_OBJECT(cell), "font", "monospace", NULL);
 	g_signal_connect_after(object, "realize",
 			       G_CALLBACK(set_column_mono_font),
 			       G_OBJECT(cell));
+			       
 	ADD_TEXT_COLUMN(_("_Owner"), COL_OWNER);
 	gtk_tree_view_column_set_sort_column_id(column, COL_OWNER);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
+	
 	ADD_TEXT_COLUMN(_("_Group"), COL_GROUP);
 	gtk_tree_view_column_set_sort_column_id(column, COL_GROUP);
-	ADD_TEXT_COLUMN(_("_Size"), COL_SIZE);
-	g_object_set(G_OBJECT(cell), "xalign", 1.0, "font", "monospace", NULL);
-	g_signal_connect_after(object, "realize",
-			       G_CALLBACK(set_column_mono_font),
-			       G_OBJECT(cell));
-	gtk_tree_view_column_set_sort_column_id(column, COL_SIZE);
-	ADD_TEXT_COLUMN(_("Last _Modified"), COL_MTIME);
-	gtk_tree_view_column_set_sort_column_id(column, COL_MTIME);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
 }
 
 /* Create the handers for the View interface */
