--- sane-backends-1.0.21/backend-orig/v4l.h	2010-12-01 15:49:58.000000000 -0800
+++ sane-backends-1.0.21/backend/v4l.h	2011-04-01 21:44:18.797102300 -0700
@@ -29,6 +29,8 @@
 #ifndef v4l_h
 #define v4l_h
 
+#include <libv4l1-videodev.h>
+
 /* Kernel interface */
 /* Only the stuff we need. For more features, more defines are needed */
 
@@ -46,7 +48,7 @@
 #define VID_TYPE_MPEG_ENCODER	2048	/* Can encode MPEG streams */
 #define VID_TYPE_MJPEG_DECODER	4096	/* Can decode MJPEG streams */
 #define VID_TYPE_MJPEG_ENCODER	8192	/* Can encode MJPEG streams */
-
+#if 0
 struct video_capability
 {
 	char name[32];
@@ -58,7 +60,6 @@
 	int minwidth;	/* Supported width */
 	int minheight;	/* And height */
 };
-
 struct video_picture
 {
 	__u16	brightness;
@@ -102,9 +103,10 @@
 /* bitmap is 1024x625, a '1' bit represents a clipped pixel */
 #define VIDEO_CLIPMAP_SIZE	(128 * 625)
 };
-
+#endif
 #define VIDEO_MAX_FRAME		32
 
+#if 0
 struct video_mbuf
 {
 	int	size;		/* Total memory to map */
@@ -132,7 +134,7 @@
 #define VIDEO_TYPE_CAMERA	2	
 	__u16 norm;			/* Norm set by channel */
 };
-
+#endif
 #define VIDIOCGCAP		_IOR('v',1,struct video_capability)	/* Get capabilities */
 #define VIDIOCGCHAN		_IOWR('v',2,struct video_channel)	/* Get channel info (sources) */
 #define VIDIOCSCHAN		_IOW('v',3,struct video_channel)	/* Set channel 	*/