diff -ur rox-filer-2015.02-original/ROX-Filer/src/main.c rox-filer-2015.02/ROX-Filer/src/main.c
--- rox-filer-2015.02-original/ROX-Filer/src/main.c	2013-06-02 03:11:26.000000000 -0500
+++ rox-filer-2015.02/ROX-Filer/src/main.c	2018-04-09 16:05:19.346963111 -0500
@@ -526,7 +526,7 @@
 
 	tooltips = gtk_tooltips_new();
 
-	if (euid == 0 || show_user)
+	if (show_user)
 		show_user_message = g_strdup_printf(_("Running as user '%s'"), 
 						    user_name(euid));
 	
