diff -ur rox-filer-2015.02-original/ROX-Filer/src/view_details.c rox-filer-2015.02/ROX-Filer/src/view_details.c
--- rox-filer-2015.02-original/ROX-Filer/src/view_details.c	2013-06-02 03:11:26.000000000 -0500
+++ rox-filer-2015.02/ROX-Filer/src/view_details.c	2018-04-10 17:00:05.825403894 -0500
@@ -1021,6 +1021,7 @@
 	GtkTreeView *treeview = (GtkTreeView *) object;
 	GtkTreeViewColumn *column;
 	GtkCellRenderer *cell;
+	GtkTreeSortable *sortable_list;
 	ViewDetails *view_details = (ViewDetails *) object;
 
 	view_details->items = g_ptr_array_new();
@@ -1039,6 +1040,7 @@
 
 	/* Sorting */
 	view_details->sort_fn = NULL;
+	sortable_list = GTK_TREE_SORTABLE(object);
 
 	gtk_tree_view_set_model(treeview, GTK_TREE_MODEL(view_details));
 	/* Do this after set_model, because that can generate this
@@ -1057,26 +1059,39 @@
 	ADD_TEXT_COLUMN(_("_Name"), COL_LEAF);
 	gtk_tree_view_column_set_sort_column_id(column, COL_LEAF);
 	gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
+
 	ADD_TEXT_COLUMN(_("_Type"), COL_TYPE);
 	gtk_tree_view_column_set_sort_column_id(column, COL_TYPE);
 	gtk_tree_view_column_set_resizable(column, TRUE);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
+
+	ADD_TEXT_COLUMN(_("_Size"), COL_SIZE);
+	g_object_set(G_OBJECT(cell), "xalign", 1.0, "font", "monospace", NULL);
+	g_signal_connect_after(object, "realize",
+			       G_CALLBACK(set_column_mono_font),
+			       G_OBJECT(cell));
+	gtk_tree_view_column_set_sort_column_id(column, COL_SIZE);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
+
+	ADD_TEXT_COLUMN(_("Last _Modified"), COL_MTIME);
+	gtk_tree_view_column_set_sort_column_id(column, COL_MTIME);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
+
 	ADD_TEXT_COLUMN(_("_Permissions"), COL_PERM);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
 	g_object_set(G_OBJECT(cell), "font", "monospace", NULL);
 	g_signal_connect_after(object, "realize",
 			       G_CALLBACK(set_column_mono_font),
 			       G_OBJECT(cell));
+
 	ADD_TEXT_COLUMN(_("_Owner"), COL_OWNER);
 	gtk_tree_view_column_set_sort_column_id(column, COL_OWNER);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
+
 	ADD_TEXT_COLUMN(_("_Group"), COL_GROUP);
 	gtk_tree_view_column_set_sort_column_id(column, COL_GROUP);
-	ADD_TEXT_COLUMN(_("_Size"), COL_SIZE);
-	g_object_set(G_OBJECT(cell), "xalign", 1.0, "font", "monospace", NULL);
-	g_signal_connect_after(object, "realize",
-			       G_CALLBACK(set_column_mono_font),
-			       G_OBJECT(cell));
-	gtk_tree_view_column_set_sort_column_id(column, COL_SIZE);
-	ADD_TEXT_COLUMN(_("Last _Modified"), COL_MTIME);
-	gtk_tree_view_column_set_sort_column_id(column, COL_MTIME);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
 }
 
 /* Create the handers for the View interface */
@@ -1464,11 +1479,14 @@
 					   GdkWindow *src, int x, int y)
 {
 	ViewDetails *view_details = (ViewDetails *) view;
+	GtkTreeModel *model;
 	GtkTreeView *tree = (GtkTreeView *) view;
 	GtkTreePath *path = NULL;
 	int i = -1;
 	gint cell_y;
 
+	model = gtk_tree_view_get_model(tree);
+
 	if (gtk_tree_view_get_path_at_pos(tree, x, y + 4, &path, NULL,
 					  NULL, &cell_y))
 	{
