diff -rupN rox-git.orig/ROX-Filer/src/menu.c rox-git/ROX-Filer/src/menu.c
--- rox-git.orig/ROX-Filer/src/menu.c	2013-06-02 18:11:26.000000000 +1000
+++ rox-git/ROX-Filer/src/menu.c	2015-11-04 13:56:48.583223841 +1000
@@ -216,7 +216,7 @@ static GtkItemFactoryEntry filer_menu_de
 {">" N_("Delete"),	    	"<Ctrl>X", file_op, FILE_DELETE, "<StockItem>", GTK_STOCK_DELETE},
 {">",				NULL, NULL, 0, "<Separator>"},
 {">" N_("Shift Open"),   	NULL, file_op, FILE_OPEN_FILE},
-{">" N_("Send To..."),		NULL, file_op, FILE_SEND_TO, NULL},
+{">" N_("Open With..."),		NULL, file_op, FILE_SEND_TO, NULL},
 {">",				NULL, NULL, 0, "<Separator>"},
 {">" N_("Set Run Action..."),	"asterisk", file_op, FILE_RUN_ACTION, "<StockItem>", GTK_STOCK_EXECUTE},
 {">" N_("Set Icon..."),		NULL, file_op, FILE_SET_ICON, NULL},
