diff -ur rox-filer-2015.02-original/ROX-Filer/src/action.c rox-filer-2015.02/ROX-Filer/src/action.c
--- rox-filer-2015.02-original/ROX-Filer/src/action.c	2015-02-12 09:39:03.000000000 -0600
+++ rox-filer-2015.02/ROX-Filer/src/action.c	2018-04-10 19:07:15.453517881 -0500
@@ -1327,10 +1327,16 @@
 	{
 		int		err;
 		gboolean	merge;
+		gboolean	is_newer;
 
 		merge = S_ISDIR(info.st_mode) && S_ISDIR(dest_info.st_mode);
 
-		if (!merge && o_newer && info.st_mtime > dest_info.st_mtime)
+		/* Using greater than or equal because people who tick the
+		 * "Newer" checkbox probably don't want to be prompted whether
+		 * to overwrite a file that has an identical mtime. */
+		is_newer = info.st_mtime >= dest_info.st_mtime;
+
+		if (!merge && o_newer && is_newer)
 		{
 			/* Newer; keep going */
 		}
@@ -1338,7 +1344,7 @@
 		{
 			printf_send("<%s", path);
 			printf_send(">%s", dest_path);
-			if (!printf_reply(from_parent, merge,
+			if (!printf_reply(from_parent, !o_newer || !is_newer,
 					  _("?'%s' already exists - %s?"),
 					  dest_path,
 					  merge ? _("merge contents")
@@ -1486,6 +1492,7 @@
 	{
 		struct stat	info;
 		int		err;
+		gboolean	is_newer;
 
 		if (mc_lstat(dest_path, &info))
 		{
@@ -1493,7 +1500,12 @@
 			return;
 		}
 
-		if (!is_dir && o_newer && info2.st_mtime > info.st_mtime)
+		/* Using greater than or equal because people who tick the
+		 * "Newer" checkbox probably don't want to be prompted whether
+		 * to overwrite a file that has an identical mtime. */
+		is_newer = info2.st_mtime >= info.st_mtime;
+
+		if (!is_dir && o_newer && is_newer)
 		{
 			/* Newer; keep going */
 		}
@@ -1501,7 +1513,7 @@
 		{
 			printf_send("<%s", path);
 			printf_send(">%s", dest_path);
-			if (!printf_reply(from_parent, TRUE,
+			if (!printf_reply(from_parent, !o_newer || !is_newer,
 				       _("?'%s' already exists - overwrite?"),
 				       dest_path))
 				return;
