--- freedesktop.org.xml.in.original	2017-09-06 20:15:45.000000000 -0500
+++ freedesktop.org.xml.in	2018-03-14 11:43:37.549544858 -0500
@@ -3489,7 +3489,7 @@
   </mime-type>
   <mime-type type="application/x-sharedlib">
     <_comment>shared library</_comment>
-    <magic priority="50">
+<!-- <magic priority="50">
       <match type="string" value="\177ELF" offset="0">
         <match type="byte" value="1" offset="5">
           <match type="little16" value="3" offset="16"/>
@@ -3504,8 +3504,9 @@
         <match type="little16" mask="030000" value="020000" offset="22"/>
       </match>
       <match type="string" mask="0xffffffff000000000000000000000000ff" value="\177ELF            \003" offset="0"/>
-    </magic>
+    </magic> -->
     <glob pattern="*.so"/>
+    <glob pattern="*.so.*"/>
   </mime-type>
   <mime-type type="application/x-shellscript">
     <_comment>shell script</_comment>
@@ -7109,13 +7110,13 @@
   <!-- 3D models and GCODEs -->
   <mime-type type="model/x.stl-binary">
     <_comment>STL 3D model (binary)</_comment>
-    <magic priority="10">
+<!-- <magic priority="10">
       <match type="byte" value="0x0" mask="0xFFFF" offset="134">
         <match type="byte" value="0x0" mask="0xFFFF" offset="184">
           <match type="byte" value="0x0" mask="0xFFFF" offset="234"/>
         </match>
       </match>
-    </magic>
+    </magic> -->
     <glob pattern="*.stl"/>
   </mime-type>
 
