--- freedesktop.org.xml.in.original	2013-09-30 05:01:23.000000000 -0500
+++ freedesktop.org.xml.in	2018-02-24 23:11:16.310341834 -0600
@@ -3055,7 +3055,7 @@
   </mime-type>
   <mime-type type="application/x-sharedlib">
     <_comment>shared library</_comment>
-    <magic priority="50">
+<!-- <magic priority="50">
       <match type="string" value="\177ELF" offset="0">
         <match type="byte" value="1" offset="5">
           <match type="little16" value="3" offset="16"/>
@@ -3070,8 +3070,9 @@
         <match type="little16" mask="030000" value="020000" offset="22"/>
       </match>
       <match type="string" mask="0xffffffff000000000000000000000000ff" value="\177ELF            \003" offset="0"/>
-    </magic>
+    </magic> -->
     <glob pattern="*.so"/>
+    <glob pattern="*.so.*"/>
   </mime-type>
   <mime-type type="application/x-shellscript">
     <_comment>shell script</_comment>
