diff -rupN ddcprobe-0.4.23-orig/lrmi.c ddcprobe-0.4.23/lrmi.c
--- ddcprobe-0.4.23-orig/lrmi.c	2006-03-22 12:59:05.000000000 +1000
+++ ddcprobe-0.4.23/lrmi.c	2014-10-21 10:31:27.130602131 +1000
@@ -51,6 +51,13 @@ OTHER DEALINGS IN THE SOFTWARE.
 #include <unistd.h>
 #include <fcntl.h>
 
+#if defined(__linux__) && !defined(TF_MASK)
+	#define TF_MASK X86_EFLAGS_TF
+	#define IF_MASK X86_EFLAGS_IF
+	#define VIF_MASK X86_EFLAGS_VIF
+	#define IOPL_MASK X86_EFLAGS_IOPL
+#endif
+
 #include "include/lrmi.h"
 #include "x86-common.h"
 
