diff -ur inkscapelite-old/src/widgets/ruler.c inkscapelite-0.36.3_patched-glib-rulers/src/widgets/ruler.c
--- inkscapelite-old/src/widgets/ruler.c	2016-04-21 13:24:54.000000000 +1000
+++ inkscapelite-0.36.3_patched-glib-rulers/src/widgets/ruler.c	2016-04-25 21:02:29.683450551 +1000
@@ -18,6 +18,7 @@
 #include "ruler.h"
 
 
+#define RULER_FONT            "-misc-dejavu sans mono-medium-r-normal--12-*-*-*-m-*-*-*" 
 #define RULER_WIDTH           14
 #define RULER_HEIGHT          14
 #define MINIMUM_INCR          5
@@ -150,8 +151,9 @@
 
   gc = widget->style->fg_gc[GTK_STATE_NORMAL];
   bg_gc = widget->style->bg_gc[GTK_STATE_NORMAL];
-  font = gtk_style_get_font(widget->style);
+  //font = gtk_style_get_font(widget->style);
 
+  font = gdk_font_load (RULER_FONT); 
   xthickness = widget->style->xthickness;
   ythickness = widget->style->ythickness;
   digit_height = font->ascent; /* assume descent == 0 ? */
@@ -441,7 +443,8 @@
 
   gc = widget->style->fg_gc[GTK_STATE_NORMAL];
   bg_gc = widget->style->bg_gc[GTK_STATE_NORMAL];
-  font = gtk_style_get_font(widget->style);
+  //font = gtk_style_get_font(widget->style);
+  font = gdk_font_load (RULER_FONT); 
   xthickness = widget->style->xthickness;
   ythickness = widget->style->ythickness;
   digit_height = font->ascent; /* assume descent == 0 ? */
