diff -rup hexchat-2.10.2-orig//src/common/hexchat.c hexchat-2.10.2/src/common/hexchat.c
--- hexchat-2.10.2-orig//src/common/hexchat.c	2015-09-15 04:39:29.654403498 +0800
+++ hexchat-2.10.2/src/common/hexchat.c	2015-09-15 04:42:11.571206403 +0800
@@ -733,7 +733,7 @@ static char defaultconf_commands[] =
 	"NAME WII\n"			"CMD quote WHOIS %2 %2\n\n";
 
 static char defaultconf_urlhandlers[] =
-		"NAME Open Link in a new Firefox Window\n"		"CMD !firefox -new-window %s\n\n";
+		"NAME Open Link in the default browser\n"		"CMD !defaultbrowser %s\n\n";
 
 #ifdef USE_SIGACTION
 /* Close and open log files on SIGUSR1. Usefull for log rotating */
