diff -ur xvkbd-3.7.orig/XVkbd-common.ad xvkbd-3.7/XVkbd-common.ad
--- xvkbd-3.7.orig/XVkbd-common.ad	2013-11-02 09:39:48.000000000 +1000
+++ xvkbd-3.7/XVkbd-common.ad	2015-12-05 16:59:13.593995663 +1000
@@ -50,13 +50,13 @@
 XVkbd*Repeater.minimumDelay: 100
 XVkbd*Repeater.decay: 500
 
-XVkbd*Font: -*-helvetica-medium-r-*-*-12-*-*-*-*-*-iso8859-1
-XVkbd*generalFont: -*-lucidatypewriter-bold-r-*-*-12-*-*-*-*-*-iso8859-1
-XVkbd*letterFont: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-1
-XVkbd*specialFont: -*-helvetica-medium-r-*-*-12-*-*-*-*-*-iso8859-1
-XVkbd*keypadFont: -*-helvetica-medium-r-*-*-10-*-*-*-*-*-iso8859-1
-XVkbd*banner.font: -*-lucida-bold-i-*-*-14-*-*-*-*-*-iso8859-1
-XVkbd*display_panel*label.font: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-1
+XVkbd*Font: -misc-dejavu sans-medium-r-normal-*-12-*-*-*-p-*-*-*
+XVkbd*generalFont: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
+XVkbd*letterFont: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
+XVkbd*specialFont: -misc-dejavu sans-medium-r-normal-*-12-*-*-*-p-*-*-*
+XVkbd*keypadFont: -misc-dejavu sans-medium-r-normal-*-10-*-*-*-p-*-*-*
+XVkbd*banner.font: -misc-dejavu sans-bold-r-normal-*-14-*-*-*-p-*-*-*
+XVkbd*display_panel*label.font: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
 
 XVkbd*beNiceToColormap: false
 XVkbd*topShadowContrast: 40
@@ -221,8 +221,8 @@
 XVkbd*completion_panel*view.list.width: 400
 XVkbd*completion_panel*view.list.height: 150
 
-XVkbd*completion_panel*label.font: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-1
-XVkbd*completion_panel*list.font: -*-lucida-medium-r-*-*-14-*-*-*-*-*-iso8859-1
+XVkbd*completion_panel*label.font: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
+XVkbd*completion_panel*list.font: -misc-dejavu sans-medium-r-normal-*-14-*-*-*-p-*-*-*
 XVkbd*completion_panel*label.label: Completion:
 XVkbd*completion_panel*bottom: chainTop
 XVkbd*completion_panel*label.right: chainLeft
@@ -268,8 +268,8 @@
 XVkbd*edit_fkey_panel*save_button.label: Save
 XVkbd*edit_fkey_panel*close_button.label: Close
 
-XVkbd*edit_fkey_panel*Label.font: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-1
-XVkbd*edit_fkey_panel*fkey_value_menu.font: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-1
+XVkbd*edit_fkey_panel*Label.font: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
+XVkbd*edit_fkey_panel*fkey_value_menu.font: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
 
 !! Dead Keys
 !
@@ -278,7 +278,7 @@
   dead_tilde,dead_abovedot
 
 XVkbd*deadkey_panel.title: xvkbd - dead keys
-XVkbd*deadkey_panel*font: -*-helvetica-medium-r-*-*-24-*-*-*-*-*-iso8859-1
+XVkbd*deadkey_panel*font: -misc-dejavu sans-medium-r-normal-*-24-*-*-*-p-*-*-*
 XVkbd*deadkey_panel*Repeater.width: 30
 XVkbd*deadkey_panel*Repeater.height: 36
 XVkbd*deadkey_panel*dead_diaeresis.label: \250
@@ -346,7 +346,7 @@
 
 !! You should set the label font if you want to make the window small.
 !
-!xvkbd*Font: -misc-fixed-medium-r-semicondensed--13-*-iso8859-1
+!xvkbd*Font: -misc-dejavu sans condensed-medium-r-narrow--13-*-*-*-p-*-*-*
 
 !! Default keyboard layout which is coded in the source
 
diff -ur xvkbd-3.7.orig/XVkbd-common.h xvkbd-3.7/XVkbd-common.h
--- xvkbd-3.7.orig/XVkbd-common.h	2014-05-24 17:51:01.000000000 +1000
+++ xvkbd-3.7/XVkbd-common.h	2015-12-05 16:55:56.913997827 +1000
@@ -34,13 +34,13 @@
 "XVkbd*Repeater.initialDelay: 600",
 "XVkbd*Repeater.minimumDelay: 100",
 "XVkbd*Repeater.decay: 500",
-"XVkbd*Font: -*-helvetica-medium-r-*-*-12-*-*-*-*-*-iso8859-1",
-"XVkbd*generalFont: -*-lucidatypewriter-bold-r-*-*-12-*-*-*-*-*-iso8859-1",
-"XVkbd*letterFont: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-1",
-"XVkbd*specialFont: -*-helvetica-medium-r-*-*-12-*-*-*-*-*-iso8859-1",
-"XVkbd*keypadFont: -*-helvetica-medium-r-*-*-10-*-*-*-*-*-iso8859-1",
-"XVkbd*banner.font: -*-lucida-bold-i-*-*-14-*-*-*-*-*-iso8859-1",
-"XVkbd*display_panel*label.font: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-1",
+"XVkbd*Font: -misc-dejavu sans-medium-r-normal-*-12-*-*-*-p-*-*-*",
+"XVkbd*generalFont: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*",
+"XVkbd*letterFont: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*",
+"XVkbd*specialFont: -misc-dejavu sans-medium-r-normal-*-12-*-*-*-p-*-*-*",
+"XVkbd*keypadFont: -misc-dejavu sans-medium-r-normal-*-10-*-*-*-p-*-*-*",
+"XVkbd*banner.font: -misc-dejavu sans-bold-r-normal-*-14-*-*-*-p-*-*-*",
+"XVkbd*display_panel*label.font: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*",
 "XVkbd*beNiceToColormap: false",
 "XVkbd*topShadowContrast: 40",
 "XVkbd*bottomShadowContrast: 80",
@@ -190,8 +190,8 @@
 "XVkbd*completion_panel*view.forceBars: true",
 "XVkbd*completion_panel*view.list.width: 400",
 "XVkbd*completion_panel*view.list.height: 150",
-"XVkbd*completion_panel*label.font: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-1",
-"XVkbd*completion_panel*list.font: -*-lucida-medium-r-*-*-14-*-*-*-*-*-iso8859-1",
+"XVkbd*completion_panel*label.font: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*",
+"XVkbd*completion_panel*list.font: -misc-dejavu sans-medium-r-normal-*-14-*-*-*-p-*-*-*",
 "XVkbd*completion_panel*label.label: Completion:",
 "XVkbd*completion_panel*bottom: chainTop",
 "XVkbd*completion_panel*label.right: chainLeft",
@@ -234,13 +234,13 @@
 "XVkbd*edit_fkey_panel*fkey_value_sep.horizDistance: 0",
 "XVkbd*edit_fkey_panel*save_button.label: Save",
 "XVkbd*edit_fkey_panel*close_button.label: Close",
-"XVkbd*edit_fkey_panel*Label.font: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-1",
-"XVkbd*edit_fkey_panel*fkey_value_menu.font: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-1",
+"XVkbd*edit_fkey_panel*Label.font: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*",
+"XVkbd*edit_fkey_panel*fkey_value_menu.font: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*",
 "XVkbd.deadkeys: dead_diaeresis,dead_acute,dead_grave,\
   dead_circumflex,dead_macron,dead_abovering,dead_doubleacute,\
   dead_tilde,dead_abovedot",
 "XVkbd*deadkey_panel.title: xvkbd - dead keys",
-"XVkbd*deadkey_panel*font: -*-helvetica-medium-r-*-*-24-*-*-*-*-*-iso8859-1",
+"XVkbd*deadkey_panel*font: -misc-dejavu sans-medium-r-normal-*-24-*-*-*-p-*-*-*",
 "XVkbd*deadkey_panel*Repeater.width: 30",
 "XVkbd*deadkey_panel*Repeater.height: 36",
 "XVkbd*deadkey_panel*dead_diaeresis.label: \\250",
diff -ur xvkbd-3.7.orig/XVkbd-jisx6002.ad xvkbd-3.7/XVkbd-jisx6002.ad
--- xvkbd-3.7.orig/XVkbd-jisx6002.ad	2013-03-18 22:04:23.000000000 +1000
+++ xvkbd-3.7/XVkbd-jisx6002.ad	2015-12-05 16:55:56.917331161 +1000
@@ -26,8 +26,8 @@
 xvkbd*row3*international: true
 xvkbd*row4*international: true
 xvkbd*row5*international: true
-xvkbd*specialFontSet: -*-helvetica-medium-r-*-*-12-*-*-*-*-*-iso8859-*,*--14-*
-xvkbd*fontSet: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-*,*--14-*
+xvkbd*specialFontSet: -misc-dejavu sans-medium-r-normal-*-12-*-*-*-p-*-*-*,*--14-*
+xvkbd*fontSet: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*,*--14-*
 xvkbd.modalKeytop: true
 
 xvkbd.NormalKeys: \
diff -ur xvkbd-3.7.orig/XVkbd-jisx6004.ad xvkbd-3.7/XVkbd-jisx6004.ad
--- xvkbd-3.7.orig/XVkbd-jisx6004.ad	2013-03-18 22:04:40.000000000 +1000
+++ xvkbd-3.7/XVkbd-jisx6004.ad	2015-12-05 16:55:56.917331161 +1000
@@ -26,8 +26,8 @@
 xvkbd*row3*international: true
 xvkbd*row4*international: true
 xvkbd*row5*international: true
-xvkbd*specialFontSet: -*-helvetica-medium-r-*-*-12-*-*-*-*-*-iso8859-*,*--14-*
-xvkbd*fontSet: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-*,*--14-*
+xvkbd*specialFontSet: -misc-dejavu sans-medium-r-normal-*-12-*-*-*-p-*-*-*,*--14-*
+xvkbd*fontSet: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*,*--14-*
 xvkbd.modalKeytop: true
 
 xvkbd.NormalKeys: \
diff -ur xvkbd-3.7.orig/XVkbd-korean.ad xvkbd-3.7/XVkbd-korean.ad
--- xvkbd-3.7.orig/XVkbd-korean.ad	2013-03-18 22:04:52.000000000 +1000
+++ xvkbd-3.7/XVkbd-korean.ad	2015-12-05 16:55:56.917331161 +1000
@@ -31,9 +31,9 @@
 xvkbd*row3*international: true
 xvkbd*row4*international: true
 xvkbd*row5*international: true
-xvkbd*specialFontSet: -*-helvetica-medium-r-*-*-12-*-*-*-*-*-iso8859-*,\
+xvkbd*specialFontSet: -misc-dejavu sans-medium-r-normal-*-12-*-*-*-p-*-*-*,\
   -*-*-medium-r-*-*-12-*-*-*-*-*-ksc5601.1987-0,*--12-*
-xvkbd*fontSet: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-*,\
+xvkbd*fontSet: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*,\
   -*-*-medium-r-*-*-12-*-*-*-*-*-ksc5601.1987-0,*--12-*
 xvkbd.modalKeytop: false
 
diff -ur xvkbd-3.7.orig/XVkbd-russian.ad xvkbd-3.7/XVkbd-russian.ad
--- xvkbd-3.7.orig/XVkbd-russian.ad	2015-02-11 22:13:19.000000000 +1000
+++ xvkbd-3.7/XVkbd-russian.ad	2015-12-05 16:55:56.917331161 +1000
@@ -15,8 +15,8 @@
 
 xvkbd.title: xvkbd - Virtual Keyboard (Russian)
 
-xvkbd*generalFont: -*-fixed-medium-r-normal-*-13-*-*-*-*-*-koi8-r
-xvkbd*letterFont: -*-fixed-medium-r-normal-*-13-*-*-*-*-*-koi8-r
+xvkbd*generalFont: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
+xvkbd*letterFont: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
 
 xvkbd*space.width: 80
 
diff -ur xvkbd-3.7.orig/XVkbd-slovene.ad xvkbd-3.7/XVkbd-slovene.ad
--- xvkbd-3.7.orig/XVkbd-slovene.ad	2013-03-18 22:06:27.000000000 +1000
+++ xvkbd-3.7/XVkbd-slovene.ad	2015-12-05 16:55:56.920664494 +1000
@@ -10,8 +10,8 @@
 !
 xvkbd.title: xvkbd - Virtual Keyboard (Slovene)
 
-xvkbd*generalFont: -*-lucidatypewriter-bold-r-*-*-12-*-*-*-*-*-iso8859-2
-xvkbd*letterFont: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-2
+xvkbd*generalFont: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
+xvkbd*letterFont: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
 
 xvkbd*Shift_L.width: 45
 xvkbd*space.width: 110
@@ -80,10 +80,10 @@
   dead_tilde,dead_abovedot,\
   dead_ogonek,dead_cedilla
 
-XVkbd*deadkey_panel*dead_caron.font: -*-helvetica-medium-r-*-*-24-*-*-*-*-*-iso8859-2
-XVkbd*deadkey_panel*dead_breve.font: -*-helvetica-medium-r-*-*-24-*-*-*-*-*-iso8859-2
-XVkbd*deadkey_panel*dead_ogonek.font: -*-helvetica-medium-r-*-*-24-*-*-*-*-*-iso8859-2
-XVkbd*deadkey_panel*dead_cedilla.font: -*-helvetica-medium-r-*-*-24-*-*-*-*-*-iso8859-2
+XVkbd*deadkey_panel*dead_caron.font: -misc-dejavu sans-medium-r-normal-*-24-*-*-*-p-*-*-*
+XVkbd*deadkey_panel*dead_breve.font: -misc-dejavu sans-medium-r-normal-*-24-*-*-*-p-*-*-*
+XVkbd*deadkey_panel*dead_ogonek.font: -misc-dejavu sans-medium-r-normal-*-24-*-*-*-p-*-*-*
+XVkbd*deadkey_panel*dead_cedilla.font: -misc-dejavu sans-medium-r-normal-*-24-*-*-*-p-*-*-*
 XVkbd*deadkey_panel*dead_caron.label: \267
 XVkbd*deadkey_panel*dead_breve.label: \242
 XVkbd*deadkey_panel*dead_ogonek.label: \262
diff -ur xvkbd-3.7.orig/XVkbd-turkish.ad xvkbd-3.7/XVkbd-turkish.ad
--- xvkbd-3.7.orig/XVkbd-turkish.ad	2013-03-18 22:07:35.000000000 +1000
+++ xvkbd-3.7/XVkbd-turkish.ad	2015-12-05 16:55:56.920664494 +1000
@@ -10,8 +10,8 @@
 !
 xvkbd.title: xvkbd - Virtual Keyboard (Turkish)
 
-xvkbd*generalFont: -*-lucidatypewriter-bold-r-*-*-12-*-*-*-*-*-iso8859-9
-xvkbd*letterFont: -*-helvetica-bold-r-*-*-12-*-*-*-*-*-iso8859-9
+xvkbd*generalFont: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
+xvkbd*letterFont: -misc-dejavu sans-bold-r-normal-*-12-*-*-*-p-*-*-*
 
 xvkbd*Shift_L.width: 45
 xvkbd*space.width: 110
@@ -80,10 +80,10 @@
   dead_tilde,dead_abovedot,\
   dead_ogonek,dead_cedilla
 
-XVkbd*deadkey_panel*dead_caron.font: -*-helvetica-medium-r-*-*-24-*-*-*-*-*-iso8859-2
-XVkbd*deadkey_panel*dead_breve.font: -*-helvetica-medium-r-*-*-24-*-*-*-*-*-iso8859-2
-XVkbd*deadkey_panel*dead_ogonek.font: -*-helvetica-medium-r-*-*-24-*-*-*-*-*-iso8859-2
-XVkbd*deadkey_panel*dead_cedilla.font: -*-helvetica-medium-r-*-*-24-*-*-*-*-*-iso8859-2
+XVkbd*deadkey_panel*dead_caron.font: -misc-dejavu sans-medium-r-normal-*-24-*-*-*-p-*-*-*
+XVkbd*deadkey_panel*dead_breve.font: -misc-dejavu sans-medium-r-normal-*-24-*-*-*-p-*-*-*
+XVkbd*deadkey_panel*dead_ogonek.font: -misc-dejavu sans-medium-r-normal-*-24-*-*-*-p-*-*-*
+XVkbd*deadkey_panel*dead_cedilla.font: -misc-dejavu sans-medium-r-normal-*-24-*-*-*-p-*-*-*
 XVkbd*deadkey_panel*dead_caron.label: \267
 XVkbd*deadkey_panel*dead_breve.label: \242
 XVkbd*deadkey_panel*dead_ogonek.label: \262
