diff --git a/converter/other/Makefile b/converter/other/Makefile
index 7aefda1..0d998f6 100644
--- a/converter/other/Makefile
+++ b/converter/other/Makefile
@@ -121,7 +121,7 @@ PORTBINARIES =  avstopam bmptopnm fitstopnm \
 		pamtowinicon pamtoxvmini \
 		pbmtopgm pdbimgtopam pfmtopam \
 	        pgmtopbm pgmtoppm ppmtopgm pnmtoddif \
-		pnmtopclxl pnmtorast \
+		pnmtopclxl pnmtops pnmtorast \
 		pnmtosgi pnmtosir pamtotga pnmtoxwd \
 		rasttopnm rlatopam sgitopnm sirtopnm srftopam sunicontopnm \
 		winicontopam xwdtopnm zeisstopnm
